### ISSUE TITLE
Mark Activity.Tags as obsolete and/or create an analyzer

### ISSUE THREAD

**Issue by msallin** [dotnet/runtime#49383]
The `Activity.Tags` property only return tags which are `KeyValuePair<string, string>` and shouldn't be used anymore. Having both without any hints about this is confusing for the customer and might lead to mistakes.

Don't use anymore:
https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L271-L280

Use:
https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L282-L289

For details see discussion with @tarekgh 
https://github.com/dotnet/runtime/pull/48722#discussion_r589087042

**Comment by stephentoub** [comment]
@tarekgh, is this really a code analyzer, or are you just considering obsoleting the property if it really shouldn't be used anymore?

**Comment by tarekgh** [comment]
It is more considering obsoletion. The analyzer will be needed only if we didn't obsolete it. 

CC @noahfalk

**Comment by menees** [comment]
Yes, please mark Tags as obsolete in .NET 8. I couldn't find anything in the .NET 7 docs about which one to use or prefer. Finally, a web search for "activity tags vs tagobjects" linked to this issue as the 7th link down. That's too much of a treasure hunt to figure out the preferred API. An [Obsolete] attribute on Tags that said to use TagObjects instead would be much better.

**Comment by jakebanks** [comment]
Thanks for raising this OP, this was a fun problem to face
